src/psyclone/tests/psyir/symbols/routinesymbol_test.py

```diff
@@ -142,3 +143,16 @@ def test_routinesymbol_str():
     assert (str(routine_symbol) ==
             "eyore: RoutineSymbol<some_type: DataTypeSymbol, pure=True, "
             "elemental=False>")
+
+
+def test_routinesymbol_copy():
+    '''
+    Test the RoutineSymbol.copy() method.
+    '''
+    asym = RoutineSymbol("a", is_elemental=True, is_pure=True)
+    new_sym = asym.copy()
+    assert new_sym is not asym
+    assert new_sym.name == asym.name
+    assert new_sym.visibility == asym.visibility
+    assert new_sym.is_elemental
+    assert new_sym.is_pure
```

Comment:
Please could you extend this with an example where the original RoutineSymbol does not have `is_elemental` and `is_pure` specified in the constructor.

Reply:
Done